lock-api/src/main/java/com/palantir/lock/client/ClientSplitLockService.java

```diff
@@ -45,24 +45,17 @@ protected RemoteLockService delegate() {
     }
 
     @Override
-    public LockRefreshToken lockAnonymously(LockRequest request) throws InterruptedException {
-        LockRefreshToken result = lock(LockClient.ANONYMOUS, request);
-        return result;
-    }
-
-    @Override
-    public LockRefreshToken lockWithClient(String client, LockRequest request)
+    public LockRefreshToken lock(String client, LockRequest request)
             throws InterruptedException {
-        LockRefreshToken result = lock(LockClient.of(client), request);
-        return result;
+        return lock(LockClient.of(client), request);
     }
 
     private LockRefreshToken lock(LockClient client, LockRequest request) throws InterruptedException {
         if (request.getBlockingMode() == BlockingMode.DO_NOT_BLOCK) {
             if (client == LockClient.ANONYMOUS) {
-                return nonBlockingClient.lockAnonymously(request);
+                return nonBlockingClient.lock(LockClient.ANONYMOUS_REMOTE, request);
```

Comment:
having to still have this splitting code kinda defeats the purpose, doesn't it?  Can we make ANONYMOUS.getClientId() return "".